### PR TITLE
Add the missing API docs for /articles

### LINKED ISSUE #18741: [API v1 docs] Articles
We are using [rswag](https://github.com/rswag/rswag) to implement API V1 docs and are missing the following Articles endpoints:

- [x] [Create new Article](https://developers.forem.com/api/v0#tag/articles/operation/createArticle)
- [x] [Published articles sorted by publish date](https://developers.forem.com/api/v0#tag/articles/operation/getLatestArticles)
- [x] [A published article by ID](https://developers.forem.com/api/v0#tag/articles/operation/getArticleById)
- [x] [Update an article](https://developers.forem.com/api/v0#tag/articles/operation/updateArticle)
- [x] [A published article by path](https://developers.forem.com/api/v0#tag/articles/operation/getArticleByPath)
- [x] [User's articles](https://developers.forem.com/api/v0#tag/articles/operation/getUserArticles)
- [x] [User's unpublished articles](https://developers.forem.com/api/v0#tag/articles/operation/getUserUnpublishedArticles)
- [x] [User's published articles](https://developers.forem.com/api/v0#tag/articles/operation/getUserPublishedArticles)
- [x] [User's all articles](https://developers.forem.com/api/v0#tag/articles/operation/getUserAllArticles)

### PR DESCRIPTION
**Pull request by fdocr**
## What type of PR is this? (check all applicable)

- [x] Documentation Update

## Description

Adds the missing `/articles` API docs to rswag specs

## Related Tickets & Documents

- closes #18741

## QA Instructions, Screenshots, Recordings

Open `swagger/v1/api_v1.json` on https://editor.swagger.com to check it out

### UI accessibility concerns?

N/A

## Added/updated tests?

- [x] Yes

## [Forem core team only] How will this change be communicated?

- [x] This change does not need to be communicated, and this is why not: Ongoing effort to document API V1 will be communicated as a whole when ready

## [optional] What gif best describes this PR or how it makes you feel?

Nancy Pelosi: article I is adopted

### CODE REVIEW

**Review by jaw6 — APPROVED** [review]

File: spec/requests/api/v1/docs/articles_spec.rb [hunk 1]
```diff
@@ -8,20 +8,86 @@
...
   let(:organization) { create(:organization) } # not used by every spec but lower times overall
   let(:tag) { create(:tag, :with_colors, name: "discuss") }
-  let(:article) { create(:article, featured: true, tags: "discuss", published: true) }
+  let(:published_article) { create(:article, featured: true, tags: "discuss", published: true) }
   let(:unpublished_aricle) { create(:article, published: false) }
   let(:api_secret) { create(:api_secret) }
   let(:user) { api_secret.user }
+  let(:user_article) { create(:article, featured: true, tags: "discuss", published: true, user_id: user.id) }
   let(:Accept) { "application/vnd.forem.api-v1+json" }
+  # let!(:sample_articles) { create_list(:article, 3) }
```
Comment (Ridhwana) [inline spec/requests/api/v1/docs/articles_spec.rb:17]: Should we remove this comment?

File: spec/requests/api/v1/docs/articles_spec.rb [hunk 2]
```diff
@@ -8,20 +8,86 @@
...
     path "/api/articles" do
+      post "Publish article" do
+        tags "articles"
+        description "This endpoint allows the client to create a new article.
+
+\"Articles\" are all the posts that users create on DEV that typically show up in the feed. They can be a blog post, a discussion question, a help thread etc. but is referred to as article within the code."
+        operationId "createArticle"
+        produces "application/json"
+        consumes "application/json"
+        parameter name: :article, in: :body, schema: {
```
Comment (Ridhwana) [inline spec/requests/api/v1/docs/articles_spec.rb:35]: Should we move this schema to the swagger_helper since we duplicate it on line 244?

**Review by Ridhwana — APPROVED** [review]
Thanks @fdocr this looks good. I also left two comments/suggestions.

File: spec/requests/api/v1/docs/articles_spec.rb (hunk 2 above)
Comment (fdocr) [reply spec/requests/api/v1/docs/articles_spec.rb]: Great suggestion, thanks! Just included it 👍🏼

**Pull request merged**